src/System.Text.RegularExpressions/tests/PrecompiledRegexScenarioTest.cs

```diff
@@ -0,0 +1,227 @@
+﻿using System;
+using System.Text.RegularExpressions;
+using RegexTestNamespace;
+using Xunit;
+
+namespace System.Text.RegularExpressionsTests
+{
+    public class PrecompiledRegexScenarioTest
+    {
+        [Fact]
+        public void TestPrecompiledRegex()
+        {
+            string text = "asdf134success1245something";
+            RegexTestClass testClass = new RegexTestClass();
+
+            Assert.Equal(1, testClass.Matches(text).Count);
+            Assert.Equal(1, testClass.Match(text).Groups[0].Captures.Count);
+            Assert.Equal(text, testClass.Match(text).Groups[0].Value);
+        }
+    }
+}
+
+namespace RegexTestNamespace
```

Comment:
Interesting. I didn't realize the plan is to use the Desktop API and rewrite the IL to work on .NET Core. I was hoping the Desktop API would be transitioned to .NET Core, either as an API in a separate assembly (e.g. something like `System.Text.RegularExpressions.Compiler`) or as a standalone tool, so that it could be used cross-platform. I had started prototyping such a tool a while back that used `Microsoft.Cci` instead of `Reflection.Emit`...

I look forward to seeing the tool opened sourced, but will be sad if it requires generating the assembly on the full framework first.